File: app/src/main/java/kim/uno/base/util/DateUtil.java
package kim.uno.base.util;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class DateUtil {

    public static final String DEFAULT_FORMAT = "yyyyMMddHHmmSSS";

    public static Date parse(String date) {
        return parse(date, DEFAULT_FORMAT);
    }

    public static Date parse(String date, String fromFormat) {

        if (date == null || date.trim().length() == 0) return null;

        Date result = null;
        try {
            SimpleDateFormat simpleDateFormat = new SimpleDateFormat(fromFormat);
            result = simpleDateFormat.parse(date);
        } catch (ParseException e) {
            LogUtil.e(e);
        }

        return result;
    }

    public static String format(String date, String toFormat) {
        return format(date, DEFAULT_FORMAT, toFormat);
    }

    public static String format(String date, String fromFormat, String toFormat) {
        return format(parse(date, fromFormat), toFormat);
    }

    public static String format(Date date, String toFormat) {
        String result = null;
        if (date != null) {
            SimpleDateFormat simpleDateFormat = new SimpleDateFormat(toFormat, Locale.US);
            result = simpleDateFormat.format(date);
        }
        return result;
    }

    public static String getNow() {
        return getNow(DEFAULT_FORMAT);
    }

    public static String getNow(String toFormat) {
        return format(new Date(), toFormat);
    }
}

File: base/src/main/kotlin/kim/uno/kotlin/base/util/DateUtil.kt
package kim.uno.kotlin.base.util

import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

object DateUtil {

    private val DEFAULT_FORMAT = "yyyyMMddHHmmssSSS"

    @JvmStatic
    fun parse(date: String?, fromFormat: String = DEFAULT_FORMAT): Date? {
        if (date == null || date.trim { it <= ' ' }.length == 0) return null

        try {
            val simpleDateFormat = SimpleDateFormat(fromFormat)
            return simpleDateFormat.parse(date)
        } catch (e: ParseException) { }

        return null
    }

    @JvmStatic
    fun format(date: String, toFormat: String): String? {
        return format(date, DEFAULT_FORMAT, toFormat)
    }

    @JvmStatic
    fun format(date: String, fromFormat: String, toFormat: String): String? {
        return format(parse(date, fromFormat), toFormat)
    }

    @JvmStatic
    fun format(date: Date?, toFormat: String): String? {
        if (date == null) return null

        try {
            val simpleDateFormat = SimpleDateFormat(toFormat, Locale.KOREA)
            return simpleDateFormat.format(date)
        } catch (e: ParseException) { }

        return null
    }

    @JvmStatic
    val now: String?
        get() = getNow(DEFAULT_FORMAT)

    @JvmStatic
    fun getNow(toFormat: String): String? {
        return format(Date(), toFormat)
    }

}